cashctrl_ledger/ledger.py

```diff
@@ -472,33 +684,41 @@ def fetch_journal(id: int) -> pd.DataFrame:
             dfs = pd.concat([fetch_journal(id) for id in collective_ids])
             collective = unnest(dfs, 'items')
 
-            # Find currency
+            # Map to account number and account currency
             cols = {'id': 'accountId', 'currencyCode': 'account_currency', 'number': 'account'}
             account_map = self._client.list_accounts()[cols.keys()].rename(columns=cols)
             collective = pd.merge(collective, account_map, 'left', on='accountId', validate='m:1')
+
+            # Identify base currency or foreign currency adjustment transactions
             base_currency = self.base_currency
-            account_in_base_currency = collective['account_currency'] == base_currency
+            is_fx_adjustment = ((collective['account_currency'] != base_currency) &
+                collective['currency'].isna() | (collective['currency'] == base_currency))
+
             amount = collective['debit'].fillna(0) - collective['credit'].fillna(0)
-            base_currency_amount = amount * collective['fx_rate']
-            foreign_amount = np.where(account_in_base_currency, base_currency_amount, amount)
+            currency = collective['account_currency']
+            base_amount = np.where(currency == base_currency, pd.NA,
+                np.where(is_fx_adjustment, amount, amount * collective['fx_rate']))
+            foreign_amount = np.where(currency == base_currency, amount * collective['fx_rate'],
+                                      np.where(is_fx_adjustment, 0, amount))
             mapped_collective = pd.DataFrame({
                 'id': collective['id'],
                 'date': collective['date'],
-                'currency': np.where(account_in_base_currency, base_currency, collective['currency']),
+                'currency': currency,
                 'account': collective['account'],
                 'text': collective['description'],
                 # TODO: Once precision() is implemented, use `round_to_precision()`
                 # instead of hard-coded rounding
-                'amount': pd.Series(foreign_amount).round(2),
-                'base_currency_amount': base_currency_amount.round(2),
+                'amount': pd.Series(foreign_amount).astype(pd.Float64Dtype()).round(2),
+                'base_currency_amount': pd.Series(base_amount).astype(pd.Float64Dtype()).round(2),
                 'vat_code': collective['taxName'],
                 'document': collective['document'],
             })
-            result = pd.concat([result, mapped_collective])
+            result = pd.concat([self.standardize_ledger_columns(result),
+                                self.standardize_ledger_columns(mapped_collective)])
 
         return self.standardize_ledger(result)
 
-    def _collective_transaction_currency_and_rate(self, entry: pd.DataFrame) -> Tuple[str, float]:
+    def _collective_transaction_currency_and_rate(self, entry: pd.DataFrame, suppress_error: bool = False) -> Tuple[str, float]:
```

Comment:
We should update a docstring for this method based on new `suppress_error` parameter and new `ValueError` Raise